src/main/java/build/buf/protovalidate/internal/constraints/ConstraintCache.java

```diff
@@ -243,7 +243,7 @@ private Env getRuleEnv(
                 Variable.RULES_NAME,
                 Decls.newObjectType(constraintMessage.getDescriptorForType().getFullName())),
             Decls.newVar(
-                Variable.RULE_NAME, DescriptorMappings.getCELType(constraintFieldDesc, forItems))));
+                Variable.RULE_NAME, DescriptorMappings.getCELType(constraintFieldDesc, false))));
```

Comment:
Explanation: The `forItems` in this context is for the field, _not_ the `rule`. We _never_ act on the nested items of the `rule` individually, so `forItems` should always be false for the `rule` type.